src/worldmap.cc

```diff
@@ -5221,17 +5221,50 @@ static int wmInterfaceRefresh()
     }
 
     // Render cities.
-    for (int index = 0; index < wmMaxAreaNum; index++) {
-        CityInfo* cityInfo = &(wmAreaInfoList[index]);
-        if (cityInfo->state != CITY_STATE_UNKNOWN) {
-            CitySizeDescription* citySizeDescription = &(wmSphereData[cityInfo->size]);
-            int cityX = cityInfo->x - wmWorldOffsetX;
-            int cityY = cityInfo->y - wmWorldOffsetY;
-            if (cityX >= 0 && cityX <= 472 - citySizeDescription->frmImage.getWidth()
-                && cityY >= 0 && cityY <= 465 - citySizeDescription->frmImage.getHeight()) {
-                wmInterfaceDrawCircleOverlay(cityInfo, citySizeDescription, wmBkWinBuf, cityX, cityY);
+    constexpr int PADDING = 100;
+
+    int viewportWidth = 472;
+    int viewportHeight = 465;
+
+    int bufferWidth = viewportWidth + PADDING * 2;
+    int bufferHeight = viewportHeight + PADDING * 2;
+
+    unsigned char* paddedBuf = (unsigned char*)malloc(bufferWidth * bufferHeight);
```

Comment:
It's a pretty significant change from the current code that allocates the screen buffer once, vs this which allocates a new ~500kB buffer every frame, and copies the whole screen onto it.  Maybe that isn't super significant for modern computers, though this'll be yet bigger when we have hires support for the world map